cmd/errorformat/main.go

```diff
@@ -161,3 +199,42 @@ func (t *TrackingWriter) WriteNL() (int, error) {
 func (t *TrackingWriter) NeedNL() bool {
 	return t.last != '\n'
 }
+
+type CheckStyleWriter struct {
+	mu    sync.Mutex
+	files map[string]*checkstyle.File
+	w     io.Writer
+}
+
+func (c *CheckStyleWriter) Write(e *errorformat.Entry) error {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if c.files == nil {
+		c.files = make(map[string]*checkstyle.File)
+	}
+	if _, ok := c.files[e.Filename]; !ok {
+		c.files[e.Filename] = &checkstyle.File{Name: e.Filename}
+	}
+	checkerr := &checkstyle.Error{
+		Column:   e.Col,
+		Line:     e.Lnum,
+		Message:  e.Text,
+		Severity: e.Types(),
+	}
+	c.files[e.Filename].Errors = append(c.files[e.Filename].Errors, checkerr)
+	return nil
+}
+
+func (c *CheckStyleWriter) Flash() error {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method CheckStyleWriter.Flash should have comment or be unexported